lib/ansible/modules/network/cli/cli_command.py

```diff
@@ -16,212 +16,107 @@
-        the number of I(retries) has been exceeded.
+        is returned.
     required: true
-  wait_for:
+  prompt:
     description:
-      - Specifies what to evaluate from the output of the command
-        and what conditionals to apply.  This argument will cause
-        the task to wait for a particular conditional to be true
-        before moving forward.   If the conditional is not true
-        by the configured retries, the task fails.
-        Note - With I(wait_for) the value in C(result['stdout']) can be accessed
-        using C(result), that is to access C(result['stdout'][0]) use C(result[0]) See examples.
-    aliases: ['waitfor']
-    version_added: "2.2"
-  match:
+      - A single regex pattern or a sequence of patterns to evaluate the expected
+        prompt from I(command).
+    required: false
+  answer:
     description:
-      - The I(match) argument is used in conjunction with the
-        I(wait_for) argument to specify the match policy.  Valid
-        values are C(all) or C(any).  If the value is set to C(all)
-        then all conditionals in the I(wait_for) must be satisfied.  If
-        the value is set to C(any) then only one of the values must be
-        satisfied.
-    default: all
-    choices: ['any', 'all']
-    version_added: "2.2"
-  retries:
+      - The answer to reply with if I(prompt) is matched.
+    required: false
+  sendonly:
     description:
-      - Specifies the number of retries a command should be tried
-        before it is considered failed.  The command is run on the
-        target device every retry and evaluated against the I(wait_for)
-        conditionals.
-    default: 10
-  interval:
-    description:
-      - Configures the interval in seconds to wait between retries
-        of the command.  If the command does not pass the specified
-        conditional, the interval indicates how to long to wait before
-        trying the command again.
-    default: 1
+      - Bool value that will send I(command) but not wait for a result.
+    type: bool
+    default: false
+    required: false
 """
 
 EXAMPLES = """
 - name: run show version on remote devices
   cli_command:
-    commands: show version
-
-- name: run show version and check to see if output contains Arista
-  cli_command:
-    commands: show version
-    wait_for: result[0] contains Arista
+    command: show version
 
-- name: run multiple commands on remote nodes
-  cli_command:
-    commands:
-      - show version
-      - show interfaces
-
-- name: run multiple commands and evaluate the output
-  cli_command:
-    commands:
-      - show version
-      - show interfaces
-    wait_for:
-      - result[0] contains Arista
-      - result[1] contains Loopback0
-
-- name: run commands and specify the output format
+- name: run command with json formatted output
   cli_command:
-    commands:
-      - command: show version
-        output: json
+    command: show version | json
```

Comment:
Can add example with `output` and `prompt` for more clarity

Reply:
If you can supply one, I'll drop it in (and add it to the tests if it's easy enough), otherwise I won't be able to get back to this for a bit.

Reply:
Agree with ganeshrn that a more complex example would help. Perhaps as simple as checking if a version matches a particular pattern and if so, return the answer of 'true'?  (Not that I know how to do that, mind you, but seems it might be helpful unless there's some other module/command that does a version check).  The following probly doesn't work but gives you an idea of what I'm trying to suggest:
- name: Verify the network device version is 16.1
  cli_command:
     - command: show version
        - prompt: "16.1"
        - answer: "true"

Reply:
You can refer ios_command module doc
```
  - name: run commands that require answering a prompt
    ios_command:
      commands:
        - command: 'clear counters GigabitEthernet0/1'
          prompt: 'Clear "show interface" counters on this interface \[confirm\]'
          answer: 'y'
        - command: 'clear counters GigabitEthernet0/2'
          prompt: '[confirm]'
          answer: "\r"
```